src/server/game/Spells/Spell.cpp

```diff
@@ -5380,7 +5380,15 @@ SpellCastResult Spell::CheckCast(bool strict, uint32* param1 /*= nullptr*/, uint
             case SPELL_EFFECT_POWER_DRAIN:
             {
                 // Can be area effect, Check only for players and not check if target - caster (spell can have multiply drain/burn effects)
-                if (m_caster->GetTypeId() == TYPEID_PLAYER)
```

Comment:
Just completely remove all of it for both SPELL_EFFECT_POWER_BURN and SPELL_EFFECT_POWER_DRAIN

Reply:
Oh, I wanted you to just remove the entire switch case for those two effects